extension/src/Xliff/XLIFFCache.ts

```diff
@@ -0,0 +1,78 @@
+import { Xliff } from "./XLIFFDocument";
+import * as path from "path";
+import { InvalidXmlError } from "../Error";
+import { logger } from "../Logging/LogHelper";
+import * as SettingsLoader from "../Settings/SettingsLoader";
+
+class XliffCache {
+  private cache: Map<string, Xliff>;
+  private enabled: boolean;
+
+  public get size(): number {
+    return this.cache.size;
+  }
+
+  public get isEnabled(): boolean {
+    return this.enabled;
+  }
+
+  constructor() {
```

Comment:
Maybe have settings as parameter in constructor, to make it more testable?
Then you can add tests for when cached is disabled... 😉